lib/patch-https-request-for-proxying.js

```diff
@@ -0,0 +1,98 @@
+/*
+ * Monkey-patch https.request to proxy request to the broker server
+ * according the the env variables `https_proxy` and `no_proxy`:
+ * get the https proxy from the env. proxy requests unless the
+ * broker server is in the no_proxy list.
+ * Start from the "Entry point" below.
+ */
+const url = require('url');
+const tunnel = require('tunnel');
+const {brokerServerUrl} = require('./config');
+const brokerServer = url.parse(brokerServerUrl || '');
+brokerServer.port = brokerServer.port ||
+  brokerServer.protocol === 'https:' ? '443' : '80';
+const httpsProxy = process.env.HTTPS_PROXY || process.env.https_proxy;
```

Comment:
I think we should read this from `require('./config')` – keep env access to a single place